impl/src/main/java/io/jsonwebtoken/impl/security/KeysBridge.java

```diff
@@ -95,10 +96,15 @@ public static byte[] findEncoded(Key key) {
         return encoded;
     }
 
-    public static boolean isSunPkcs11GenericSecret(Key key) {
-        return key instanceof SecretKey &&
-                key.getClass().getName().equals(SUNPKCS11_GENERIC_SECRET_CLASSNAME) &&
-                SUNPKCS11_GENERIC_SECRET_ALGNAME.equals(key.getAlgorithm());
+    public static boolean isGenericSecret(Key key) {
```

Comment:
Because this method previously reflected that it was a PKCS11 key (not just secret), please check to ensure that any calling logic does not reflect PKCS11 needs/context.

If not, and all we care about is a generic secret, then we can probably remove the `SUNPKCS11_GENERIC_SECRET_CLASSNAME` check and keep the `isGenericSecret` rename.

Reply:
Yeah this had only one use from the DefaultMacAlgorithm. Will check if the comment there needs updating.